### ISSUE TITLE
[Android] IsReadOnly=true EditText controls are not focusable

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#6250]
### Description

When Entry, Editor, and SearchBar controls are set with IsReadOnly=true, they are not only ReadOnly but also not focusable on Android. We should disassociate the two so that the focus behavior is not affected, and we can correct related keyboard accessibility anomalies.

The following code must be removed:
https://github.com/dotnet/maui/blob/523cb3eca9bf0e3c7b63c5748f36b5e570c1c829/src/Core/src/Platform/Android/EditTextExtensions.cs#L150-L151
https://github.com/dotnet/maui/blob/523cb3eca9bf0e3c7b63c5748f36b5e570c1c829/src/Core/src/Platform/Android/EditTextExtensions.cs#L170-L171
https://github.com/dotnet/maui/blob/523cb3eca9bf0e3c7b63c5748f36b5e570c1c829/src/Core/src/Platform/Android/SearchViewExtensions.cs#L96-L97

and a solution must replace it. One potential approach may be to nest the control within a container which is focusable.

### Steps to Reproduce

1. Run .NET MAUI control gallery
2. Navigate to EntryPage, EditorPage, SearchBarPage
3. Use keyboard to tab through controls. Notice the keyboard cannot reach the IsReadOnly=true control

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

/

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.6.0 Preview 4.0. Repro on Android emulator (13.0-API 33) with below Project:
[6250.zip](https://github.com/dotnet/maui/files/11292046/6250.zip)

**Comment by jaosnz-rep** [comment]
Also reproduced on iOS, Mac, and Windows platforms.

**Comment by Zhanglirong-Winnie** [comment]
This issue reproduces on Android, iOS, and MacCatalyst platforms. It works fine on Windows platforms.
![Image](https://github.com/user-attachments/assets/26cd82ff-2c62-477e-b3a9-dae3a62864fb)
